app/controllers/answers_controller.rb

```diff
@@ -0,0 +1,32 @@
+class AnswersController < ApplicationController
+  def new
+    @answer = Answer.new
+  end
+
+  def create
+    answer = Answer.new(answer_params)
+    if answer.save
+      question = Question.find_by(id: answer.question_id)
+      request.xhr? ? render(partial: 'answer', layout: false, locals: {answer: answer}) : redirect_to(question)
```

Comment:
question is already available as answer.question - you don't need to explicitly load it here.